File: typescript/src/v2/chromatic_circle/mixers/BlueMixer.ts
import { Color } from '../Color';
import { ColorMixer } from './ColorMixer';

export class BlueMixer implements ColorMixer {
  private readonly color: Color;

  constructor() {
    this.color = Color.BLUE;
  }

  public canMix(color: Color): boolean {
    return color === this.color;
  }

  public mixColors(color: Color): Color {
    if (color == this.color) {
      return color;
    }
    if (color === Color.YELLOW) {
      return Color.GREEN;
    }
    if (color === Color.RED) {
      return Color.VIOLET;
    }
    throw new Error(`invalid color: ${color}`);
  }
}

File: java/src/main/java/com/circulocromatico/BlueMixer.java
package com.circulocromatico;

public class BlueMixer implements ColorMixer {
    public static final ChromaticCircle.Color BLUE = ChromaticCircle.Color.BLUE;

    @Override
    public ChromaticCircle.Color mixColor(ChromaticCircle.Color color) {
        if (BLUE.equals(color)) {
            return color;
        }
        if (color == ChromaticCircle.Color.YELLOW) {
            return ChromaticCircle.Color.GREEN;
        }
        if (color == ChromaticCircle.Color.RED) {
            return ChromaticCircle.Color.VIOLET;
        }
        throw new IllegalArgumentException("Invalid color: " + color);
    }
    @Override
    public boolean canMix(ChromaticCircle.Color color) {
        return color == BLUE;
    }
}